src/context/UserContext.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  createContext,
+  useContext,
+  useReducer,
+  ReactNode,
+  Dispatch,
+} from 'react';
+
+interface UserState {
+  name: string;
+  email: string;
+  isLoggedIn: boolean;
+  currency: string;
+  language: string;
+  theme: string;
+  notificationSettings: {
+    expenseAlert: boolean;
+    budget: boolean;
+    tipsArticle: boolean;
+  };
+}
+
+type UserAction =
+  | { type: 'SET_USER'; payload: Partial<UserState> }
+  | { type: 'LOG_IN' }
+  | { type: 'LOG_OUT' }
+  | { type: 'SET_LANGUAGE'; payload: string }
+  | { type: 'SET_THEME'; payload: string }
+  | { type: 'SET_CURRENCY'; payload: string }
+  | {
+      type: 'TOGGLE_NOTIFICATION';
+      payload: 'expenseAlert' | 'budget' | 'tipsArticle';
+    };
+
+const initialState: UserState = {
+  name: 'john',
+  email: 'johndoe@gmail.com',
+  isLoggedIn: false,
+  currency: 'USD',
+  language: 'en',
+  theme: 'light',
+  notificationSettings: {
+    expenseAlert: false,
+    budget: false,
+    tipsArticle: false,
+  },
+};
+
+const UserContext = createContext<UserState | undefined>(undefined);
+const UserDispatchContext = createContext<Dispatch<UserAction> | undefined>(
+  undefined,
+);
+
+function userReducer(state: UserState, action: UserAction): UserState {
+  switch (action.type) {
+    case 'SET_USER':
```

Comment:
After adding enum for action type, use enum here, e.g. `case UserActionType.SET_USER:`

Reply:
Fixed